scripts/gen_dictionaries.py

```diff
@@ -154,7 +145,12 @@ def GetFooter():
                 #:CL_MEM_READ_ONLY: <<CL_MEM_READ_ONLY,{CL_MEM_READ_ONLY_label}>>
                 #:CL_MEM_READ_ONLY_anchor: [[CL_MEM_READ_ONLY]]{CL_MEM_READ_ONLY}
                 linkFile.write('// ' + name + '\n')
-                linkFile.write(':' + name + '_label: pass:q[`' + sName + '`]\n')
+                linkFile.write('ifdef::basebackend-html[]\n')
+                linkFile.write(':' + name + '_label: pass:q[`' + htmlName + '`]\n')
+                linkFile.write('endif::basebackend-html[]\n')
+                linkFile.write('ifdef::backend-pdf[]\n')
```

Comment:
Should this be ifndef::basebackend-html instead? Asciidoctor has backends besides just the HTML and PDF ones, and with the preprocessing above, nothing would be output for those backends, should we choose to use them. Replacing with &#8203; for all backends except the HTML one would at least be in keeping with what we already had.

Reply:
Yes, good point.  As I mentioned on #393 this didn't quite work out when I tried it though, and I think I figured out why.  In short, the basebackend for the PDF converter is also HTML, which is a little confusing.  See:

https://github.com/asciidoctor/asciidoctor/issues/2479

I've changed the script to instead look for the HTML5 **backend**, which I think has the desired behavior:

```asciidoc
// CL_CHAR_BIT
ifdef::backend-html5[]
:CL_CHAR_BIT_label: pass:q[`CL_<wbr>CHAR_<wbr>BIT`]
endif::[]
ifndef::backend-html5[]
:CL_CHAR_BIT_label: pass:q[`CL_&#8203;CHAR_&#8203;BIT`]
endif::[]
```

This should use the `<wbr>` syntax for our current HTML output, but will use `&#8203;` for everything else.  Does this seem better?  Thanks!

Reply:
Looks good.

Reply:
Does that mean that in this earlier version, when using the PDF backend, _both_ htmlName _and_ otherName were getting written? I.e. the recent change fixing things for other backends, but _also_ fixes the output for the PDF backend?

Reply:
> Does that mean that in this earlier version, when using the PDF backend, _both_ htmlName _and_ otherName were getting written?

Yes, I believe so.  In the previous version, the attribute was first defined to use `<wbr>` because the PDF backend uses the HTML basebackend, but then it was re-defined to use `&#8203;` for the PDF backend.  This means that it worked for both the HTML and PDF backends, but a bit confusingly, and if the two conditions were switched it would break unexpectedly.

The updated version is much more robust.  Thanks again!